### DISCUSSION TITLE
Decode data by publish()

### DISCUSSION THREAD

**Discussion by stanleyseow** [reach-sh/reach-lang#1499] [General]
How to decode the data in the EVM blockchain / explorer , both published numbers are UInt ( uint256 in ETH ) ?

```
S.only(() => { 
    const sellPrice = declassify(interact.getSellprice() ); 
    const tPrice = declassify(interact.getTprice() ); 
  });

  S.publish(sellPrice, tPrice);
  commit();
```

This is the transaction data ( https://mumbai.polygonscan.com/tx/0x7ee61cb9f01da81e0d95e99e245b50c57c8e4559d19bdd888e90c57ff4d8a918 )

Actual data :

0x873779a10000000000000000000000000000000000000000000000000000000001b89e5300000000000000000000000000000000000000150a9506967b8b9402866c0000

**Comment by jeapostrophe** [comment]
If you look at the ABI spec (in the generated `.mjs` build file, you'll see everything. But, basically, the actual publish has 1 extra number at the front. The extra number is the block time (lastConsensusTime) and the other two are your inputs, in order
